# HG changeset patch
# User Max Brister <max@2bass.com>
# Date 1347149157 21600
#      Sat Sep 08 18:05:57 2012 -0600
# Node ID 5187c3ea5a687ef4cf5c016b7ba1e8977c86c6c6
# Parent  8125773322d4b4f35fd6e3ed2034b7ece89c4f56
run-octave.in: Add -gud2 option to support new versions of gud

diff --git a/run-octave.in b/run-octave.in
--- a/run-octave.in
+++ b/run-octave.in
@@ -55,16 +55,21 @@ INFOFILE="$top_srcdir/doc/interpreter/oc
 if [ $# -gt 0 ]; then
   if [ "x$1" = "x-g" ]; then
     driver="gdb --args"
     shift
   elif [ "x$1" = "x-gud" ]; then
     ## Frontends for gdb (e.g. Emacs's GUD mode) need --annotate=3
     driver="gdb --annotate=3 --args"
     shift
+  elif [ "x$1" = "x-gud2" ]; then
+    ## The latest version of gud needs -i=mi. There isn't a good way to check
+    ## this at configure time, so we just add a gdb2 flag
+    driver="gdb -i=mi --args"
+    shift
   elif [ "x$1" = "x-valgrind" ]; then
     driver="valgrind --tool=memcheck"
     shift
   elif [ "x$1" = "x-strace" ]; then
     driver="strace -o octave.trace"
     shift
   elif [ "x$1" = "x-cli" ]; then
     octave_executable="$builddir/src/octave-cli"
